src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -294,6 +293,14 @@ private SelectItem GenerateSelectItem(SelectTermToken tokenIn)
 
             IEdmTypeReference elementTypeReference = targetElementType.ToTypeReference();
 
+            // When Creating Range Variables, we only need a Navigation Source when the elementTypeReference is a StructuredTypeReference.
+            // When the elementTypeReference is NOT StructuredTypeReference, We will create a NonResourceRangeVariable which don't require a Navigation Source.
+            if (elementTypeReference != null && elementTypeReference.IsStructured())

```

Comment:
```suggestion
            if (elementTypeReference?.IsStructured())
```
Let's use the cool toys now that we have them :)

Reply:
This is breaking my tests.
`System.InvalidOperationException : Nullable object must have a value.`